tests/cbmc/sources/make_common_datastructures.c

```diff
@@ -772,8 +772,8 @@ void cbmc_populate_s2n_connection(struct s2n_connection *s2n_connection)
     s2n_connection->recv                     = malloc(sizeof(*(s2n_connection->recv))); /* Function pointer. */
     s2n_connection->send_io_context          = malloc(sizeof(*(s2n_connection->secret_cb)));
     s2n_connection->recv_io_context          = malloc(sizeof(*(s2n_connection->secret_cb)));
-    cbmc_populate_s2n_crypto_parameters(&(s2n_connection->initial));
-    cbmc_populate_s2n_crypto_parameters(&(s2n_connection->secure));
+    cbmc_populate_s2n_crypto_parameters(s2n_connection->initial);
+    cbmc_populate_s2n_crypto_parameters(s2n_connection->secure);
```

Comment:
Why wasn't this being caught by our CI? We changed the crypto parameters to pointers a while back.

Reply:
Apparently it still passes the tests, even though the types are off. CBMC didn't check type compatibility between compilation units when linking until a recent version. We were looking to update s2n-tls to a more recent CBMC for CI as we have much better contract support running now that we would like to try, but it needs a recent CBMC, which triggered this.

Reply:
It would be good to get your approval for this so we can upgrade to the new and improved CBMC.

Reply:
Interesting. I really wonder how CBMC was functioning using a pointer to a pointer where it expected a pointer though :\